### ISSUE TITLE
Feature request: Reporting of missed stimulus presentation deadlines

### ISSUE THREAD

**Issue by danmyles** [Psychtoolbox-3/Psychtoolbox-3#773]
TLDR: It would be useful if PTB reported missed deadlines for `Screen('Flip', window)` and `Screen('Flip', window, [some target time])` separately.

At the end of a PTB session, PTB prints a message along the line of the following:

```
INFO: PTB's Screen('Flip', 10) command seems to have missed the requested stimulus presentation deadline
INFO: a total of 13 times out of a total of 1500 flips during this session.
```

Typically in designing a task there will be numerous displays for which timing is not important. For example, I may present a series of instructions to a participant prior to a task. I would typically schedule these flips using Screen('Flip', window) to target the next available frame, rather than specify [when], as the timing is not important.  It can still be useful for PTB to report timing issues for these flips, as it provides a general diagnostic of the system performance. But...

Then there are the moments that really matter. I may want to precisely control the rate of stimulus presentation, or need tight control over some ISI etc. In these cases timing does matter and if PTB + the system misses these targets I really want to know about them so I can debug my code.

It would be more useful if PTB reported each of these types of failures separately at the end of a session. For instance a message along the lines of:

```
INFO: PTB's Screen('Flip', 10) command seems to have missed the requested stimulus presentation deadline
INFO: a total of 13 times out of a total of 1500 flips during this session.
INFO: a subtotal of 3 times out of 500 scheduled flips (i.e. missed target [, when]).
INFO: a subtotal of 10 times out of 1000 unscheduled flips (i.e. no [, when] - next available frame scheduled).
```

Why do this? -> If most of the missed deadlines are occurring on my scheduled flips, it's probably my code. If they also occur on unscheduled flips at a reasonably high rate I would suspect my system is performing poorly.

**Comment by kleinerm** [comment]
The recommended way to make sure that timing is correct when it matters would be to evaluate the various return values of ``Screen('Flip/AsyncFlipCheck/AsyncFlipEnd')`` etc., e.g., compare ``VBLTimestamp`` or ``StimulusOnsetTime`` against your own model of what should happen, as demonstrated, e.g., in VBLSyncTest.m. Or use the ``missed`` flag as a slightly less foolproof indicator on a flip by flip basis. Whenever that ``missed`` value is greater than zero, it gets counted as missed flip for the summary reporting at the end of a session that you mentioned.

I can see however that a more detailed summary like you propose could be useful for some quick sanity check. If you buy a paid support membership ``help PsychPaidSupportAndServices`` and post the authentication token here as proof of purchase i would count this as a vote for this feature and have a closer look if this can be easily implemented for a future release - no guarantees though that it will happen quickly or at all, as a single paid membership may or may not pay enough for the work time of implementing it.
